exporter/datasetexporter/factory.go

```diff
@@ -39,45 +36,19 @@ func NewFactory() exporter.Factory {
 
 func createDefaultConfig() component.Config {
 	return &Config{
-		MaxDelayMs:      maxDelayMs,
+		MaxDelay:        maxDelay,
+		TracesSettings:  NewDefaultTracesSettings(),
 		RetrySettings:   exporterhelper.NewDefaultRetrySettings(),
 		QueueSettings:   exporterhelper.NewDefaultQueueSettings(),
 		TimeoutSettings: exporterhelper.NewDefaultTimeoutSettings(),
 	}
 }
 
-func createLogsExporter(ctx context.Context, set exporter.CreateSettings, config component.Config) (exporter.Logs, error) {
-	cfg := config.(*Config)
-	e, err := getDatasetExporter("logs", cfg, set.Logger)
-	if err != nil {
-		return nil, fmt.Errorf("cannot get DataSetExpoter: %w", err)
-	}
-
-	return exporterhelper.NewLogsExporter(
-		ctx,
-		set,
-		config,
-		e.consumeLogs,
-		exporterhelper.WithQueue(cfg.QueueSettings),
-		exporterhelper.WithRetry(cfg.RetrySettings),
-		exporterhelper.WithTimeout(cfg.TimeoutSettings),
-	)
-}
-
-func createTracesExporter(ctx context.Context, set exporter.CreateSettings, config component.Config) (exporter.Traces, error) {
-	cfg := config.(*Config)
-	e, err := getDatasetExporter("traces", cfg, set.Logger)
-	if err != nil {
-		return nil, fmt.Errorf("cannot get DataSetExpoter: %w", err)
+// castConfig casts it to the Dataset Config struct.
+func castConfig(c component.Config) *Config {
+	cfg, ok := c.(*Config)
+	if !ok {
+		panic("config structure is not of type *datasetexporter.Config")
```

Comment:
no panics please

Reply:
Most of the components have just `conf := cfg.(*Config)` - and if incorrect parameter is used it panics anyways. So it's indeed little bit redundant.